src/main/java/org/springframework/data/elasticsearch/core/convert/MappingElasticsearchConverter.java

```diff
@@ -354,6 +354,9 @@ public void afterPropertiesSet() {
 		Collection<Object> target = createCollectionForValue(targetType, source.size());
 
 		for (Object value : source) {
+			
+			if(null == value)
+				return null;
```

Comment:
```suggestion
			if(value == null) {
				return null;
			}
		
```
We do the comparison the other way round and always use the curly braces.